apps/dashboard/src/components/editor-overlays.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { WorkflowResponseDto } from '@novu/shared';
+
+import { EditTranslationPopover } from '@/components/workflow-editor/steps/email/translations/edit-translation-popover/edit-translation-popover';
+import { PayloadSchemaDrawer } from '@/components/workflow-editor/payload-schema-drawer';
+import { IsAllowedVariable, LiquidVariable } from '@/utils/parseStepVariables';
+
+type EditorOverlaysProps = {
+  // Translation-related props
+  isTranslationPopoverOpen?: boolean;
+  selectedTranslation?: { translationKey: string; from: number; to: number } | null;
+  onTranslationPopoverOpenChange?: (open: boolean) => void;
+  onTranslationDelete?: () => void;
+  onTranslationReplaceKey?: (newKey: string) => void;
+  translationTriggerPosition?: { top: number; left: number } | null;
+
+  // Variable and schema-related props
+  variables: LiquidVariable[];
+  isAllowedVariable: IsAllowedVariable;
+
+  // Workflow and schema-related props
+  workflow?: WorkflowResponseDto;
+  isPayloadSchemaDrawerOpen: boolean;
+  onPayloadSchemaDrawerOpenChange: (open: boolean) => void;
+  highlightedVariableKey?: string | null;
+
+  // Feature flags
+  enableTranslations?: boolean;
+};
+
+export function EditorOverlays({
```

Comment:
This is the component that is used in the `control-input` and `email-body` (HTML/Maily). The main purpose is to show the translation popover or schema drawer. It was moved outside from the `variable-editor.tsx`